datastore/storage/boltdb.go

```diff
@@ -0,0 +1,151 @@
+package storage
+
+import (
+	"errors"
+	"os"
+	"strconv"
+
+	"github.com/boltdb/bolt"
+)
+
+type DatastoreBoltDB struct {
+	db      *bolt.DB
+	lastErr error
+}
+
+// NewBoltDBStorage Creates a new BoltDB storage
+func NewBoltDBStorage() *DatastoreBoltDB {
+
+	ds := &DatastoreBoltDB{}
+
+	fname := os.Getenv("WASMVISION_STORAGE_BOLTDB_FILENAME")
+	if fname == "" {
+		ds.lastErr = errors.New("WASMVISION_STORAGE_BOLTDB_FILENAME env var not defined")
+		return ds
+	}
+
+	fmodeInt := 0644
+	fmode := os.Getenv("WASMVISION_STORAGE_BOLTDB_FILEMODE")
+	if fmode != "" {
+		i, err := strconv.ParseInt(fmode, 8, 32)
+		if err == nil {
+			fmodeInt = int(i)
+		}
+	}
+
+	db, err := bolt.Open(fname, os.FileMode(fmodeInt), nil)
+	if err != nil {
+		ds.lastErr = err
+		return ds
+	}
+
+	return &DatastoreBoltDB{
+		db: db,
+	}
+}
+
+// Error returns last operational error
+func (ds *DatastoreBoltDB) Error() error {
+	return ds.lastErr
+}
+
+// Get returns a data value from the store.
+func (ds *DatastoreBoltDB) Get(root string, key string) (string, bool) {
+
+	value := ""
+
+	err := ds.db.View(func(tx *bolt.Tx) error {
```

Comment:
```suggestion
	ds.lastErr = ds.db.View(func(tx *bolt.Tx) error {
```

Could do this and remove the assignment in line 66?

Reply:
done